plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy

```diff
@@ -136,33 +130,46 @@ class QuiltPackage {
         this.setup()
     }
 
+    Namespace packageNamespace() {
+        S3PhysicalKey registryPath = new S3PhysicalKey(bucket, '', null)
+        Registry registry = new Registry(registryPath)
+        Namespace namespace = registry.getNamespace(packageName)
+        return namespace
+    }
+
+    String resolveHash(Namespace namespace) {
+        if (hash == 'latest' || hash == null || hash == 'null') {
+            return namespace.getHash('latest')
+        }
+        return hash
+    }
+
+    Manifest packageManifest() {
+        Namespace namespace = packageNamespace()
+        String newHash = resolveHash(namespace)
+        Manifest manifest = namespace.getManifest(newHash)
+        return manifest
+    }
+
     /**
      * Returns {@code List<String>} of object keys below a subpath.
      *
-     * <p> Because the `quilt3` CLI does not provide a direct way to list
-     * the logical keys ("files") inside a package, we have to infer it
-     * from the actual files inside the install folder.
-     *
-     * <p> To do this, we list the full path of the files directly inside
-     * the subfolder, then remove the top-level folder ("base") to get
-     * the relative keys.
-     *
      * @param   subpath
      *          folder inside the package (use '' for top-level)
      *
      * @return  List of the child object keys, as Strings
      */
+
     List<String> relativeChildren(String subpath) {
-        Path subfolder = folder.resolve(subpath)
-        String base = subfolder.toString() + '/'
-        List<String> result = []
-        final String[] children = subfolder.list().sort()
-        //log.debug("relativeChildren[${base}] $children")
-        for (String pathString : children) {
-            def relative = pathString.replace(base, '')
-            result.add(relative)
+        Manifest manifest = packageManifest()
+        // get names of entries
+        List<String> children = manifest.getEntries().keySet().collect { String key ->
+            if (key.startsWith(subpath)) {
```

Comment:
Make sure that this handles "/" properly? I.e., that filtering for "/a" includes "/a/b", but not "/ab"?

Reply:
Can you clarify?  Is a leading "/" required? Do we only require the full folder for absolute subpaths?
Specifically, what's the expected result for these subpaths when the package contains ".ipynb_checkpoints/data_products-checkpoint.ipynb"?

```
        subpath               | expected_size
        ''                    | 8
        '.ipynb_checkpoints'  | 1
        '/.ipynb_checkpoints' | 1
        '.ipynb'              | 1
        '/.ipynb'             | 0
```

Reply:
I am very confused by this. I rewrote the test to make sense for me, which is:
```
        where:
        subpath               | expected_size
        ''                    | 8
        '.ipynb_checkpoints'  | 1
        '.ipynb_checkpoints/' | 1
        '.ipynb'              | 1
        '.ipynb/'             | 0
```